### DISCUSSION TITLE
Adding a method for cropping based on fractional coords instead of pixel coords

### DISCUSSION THREAD

**Discussion by ancri** [python-pillow/Pillow#6272] [Ideas]
Greetings! I'd find it very useful to have an Image method similar to `crop` but which takes fractions rather than pixel coordinates, so as to play more nicely (and predictably!) with images of varying sizes, but a known aspect ratio. The proposed method would just convert the fractions to pixels and then call `crop` under the hood.

Would anyone else find this useful? Is this something the devs would be open to adding?

Example: 
if `img.size == (1920, 1080)` then 
`img.cropf((0.1, 0.65, 0.2, 0.75)) == img.crop((192, 702, 384, 810))`.

(where `cropf` is the name of the new proposed method)

**Comment by radarhere** [comment]
I feel like this function would be an odd exception to the general co-ordinate system of Pillow.

If it made it in, then why stop with `crop()`? Users might also want to `resize()` to fractions (although admittedly, `reduce()` does partly fill this role), or `thumbnail()` to fractions. My opinion is that it is better to stick to pixels.

**Reply by ancri** [reply]
Good point, it would indeed seem like a bizarre exception to have this only apply to cropping but nothing else (though, consistency aside, in my experience whenever I'm resizing I generally have a specific absolute size I'm targeting, whereas when I'm cropping I'm usually thinking in relative terms e.g. "this part of the image" irrespective of its pixel size... but still I agree it would feel bizarre).

What if this were implemented as a new argument that can be passed to all of the normally pixel-based methods (`crop`, `resize`, and so on), let's call it `fractional_coordinates`, defaulting to False, but if passed as True would interpret the coordinates as fractional?

**Reply by radarhere** [reply]
I don't personally see the need for this to be bundled into Pillow. It's a piece of functionality that should be straightforward for users to implement themselves, but that would complicate many methods if implemented properly within this library. Pillow has a [co-ordinate system](https://pillow.readthedocs.io/en/stable/handbook/concepts.html#coordinate-system), and I'm reluctant to complicate it.

**Reply by ancri** [reply]
That's fair thing to be reluctant about. It's very unfortunate though, as this means that I have to continue to include that piece of functionality as an additional section of code separately in every application.
